### PR TITLE
Training details and publish page

### PR DESCRIPTION
**Pull request by edwardhorsford**
Key changes:
* Removes the `Type of training` section and replaces with `Training details` section
* Now optionally collects a commencement date
* Collecting training route is now part of `Programme details`
* If a provider has courses in Publish, these are an option rather than providing programme details manually

## Training details
Screenshot 2020-11-20 at 17 44 23

## Course selection
Screenshot 2020-11-20 at 17 44 47

## Course confirmation
Screenshot 2020-11-20 at 17 45 06

### CODE REVIEW

**Comment by edwardhorsford** [comment]
Things that don't work yet:
* Modifying the commencement date on existing records
* Editing the programme details
* Making changes to programme details after a record is submitted for trn

File: app/filters/strings.js [hunk 1]
```diff
@@ -35,6 +35,19 @@ filters.isString = str => {
   return _.isString(str)
 }
 
+// Assessment only => an Assesment only
+// Provider-led => a provider led
+filters.prependWithAOrAn = string => {
```
Comment (jesseyuen) [inline app/filters/strings.js:40]: 🤯

File: app/views/_includes/summary-cards/programme-details/publish-course.html [hunk 2]
```diff
@@ -61,14 +61,14 @@
           visuallyHiddenText: "subject"
         }
       ]
-    } if canAmend
+    } if canAmend and false and false
```
Comment (jesseyuen) [inline app/views/_includes/summary-cards/programme-details/publish-course.html]: if canAmend `and false and false`. What do each of the 'false' refer to? Without knowing what is going on here, could it be more meaningful?

Comment (edwardhorsford) [reply app/views/_includes/summary-cards/programme-details/publish-course.html]: The second `and false` is a typo. Removed. The first `and false` just forces the section to be disabled - but easily reenabled later if we want it.

**Review by jesseyuen — APPROVED** [review]

**Pull request merged**